### PR TITLE
Johnson Algorithm added in Java

### PR DESCRIPTION
**Pull request by Riyazul555**
Closes issue #12

### CODE REVIEW

File: Johnson/Johnson.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+// This code is contributed by Riyazul555
+
+// Time Complexity  = O(V * E + V * (V + E) * log(V))  where where V is the number of vertices and E is the number of edges
```
Comment (iuliagroza) [inline Johnson/Johnson.java:3]: See complexity discussion on #73.

File: Johnson/Johnson.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+// Space Complexity = O(V + E)                         where where V is the number of vertices and E is the number of edges
```
Comment (iuliagroza) [inline Johnson/Johnson.java:4]: see #73

File: Johnson/Johnson.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            this.first = first;
+            this.second = second;
+        }
+    }
+
+    static void bf(int nod, int n, int[] h, ArrayList<Pair>[] G, boolean[] ok) {
+        Queue<Integer> Q = new PriorityQueue<>();
+        boolean[] in = new boolean[nMax];
+        int[] nr = new int[nMax];
+        Arrays.fill(h, 1_000_000_000);
```
Comment (iuliagroza) [inline Johnson/Johnson.java:33]: see #73 for discussion on const variable usage

**Comment by Riyazul555** [comment]
Fixed @iuliagroza

File: Johnson/Johnson.java [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // This code is contributed by Riyazul555
 
-// Time Complexity  = O(V * E + V * (V + E) * log(V))  where where V is the number of vertices and E is the number of edges
-// Space Complexity = O(V + E)                         where where V is the number of vertices and E is the number of edges
+// Time Complexity  = O(V²*logv + VE)
```
Comment (iuliagroza) [inline Johnson/Johnson.java:3]: Add legend.

**Comment by Riyazul555** [comment]
Fixed @iuliagroza

**Comment by Riyazul555** [comment]
@iuliagroza please check this PR

**Review by iuliagroza — APPROVED** [review]

**Pull request merged**